sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java

```diff
@@ -1015,7 +1017,23 @@ public PDone apply(PCollection<TableRow> input) {
           table.setProjectId(options.getProject());
         }
         String jobIdToken = UUID.randomUUID().toString();
-        String tempFilePrefix = options.getTempLocation() + "/BigQuerySinkTemp/" + jobIdToken;
+        String tempLocation = options.getTempLocation();
```

Comment:
You should rely on IOChannelUtils / IOChannelFactory and not call out to paths directly.
For example:
IOChannelFactory factory = IOChannelUtils.getFactory(path);
String tempFilePrefix = factory.resolve(factory.resolve(tempLocation, "BigQuerySinkTemp"), jobIdToken);
checkArgument(testBigQueryService == null || factory instanceof GCSIOChannelFactory, ....);

Other than this comment I think the code is fine.